app/src/app/components/providers/ThemeProvider.tsx

```diff
@@ -15,31 +25,70 @@ export const useTheme = () => {
   return context;
 };
 
-export const ThemeProvider = ({ children, defaultTheme = "light", ...props }: ThemeProviderProps) => {
+const getSystemTheme = (): Theme => {
+  if (window.matchMedia("(prefers-color-scheme: dark)")?.matches) {
+    return "dark";
+  } else if (window.matchMedia("(prefers-color-scheme: light)")?.matches) {
+    return "light";
+  } else {
+    return privilegedTheme;
+  }
+};
+
+export const ThemeProvider = ({ children, ...props }: ThemeProviderProps) => {
+  const { brandingConfig, isLoading, error } = useGetBrandingConfig();
+
+  if (error) return <ErrorComponent message={error.message} error={error} />;
```

Comment:
good to structure so easy to find things.... useStates at top, then variables, then functions, then useEffects and then return statements